src/Handler/CreateAndFetch.php

```diff
@@ -51,30 +52,38 @@ final class CreateAndFetch implements RequestHandlerInterface
      */
     private $identifierGenerator;
 
+    /**
+     * @var RequestIdInjector
+     */
+    private $requestIdInjector;
+
     public function __construct(
         ExecuteCommand $writeAction,
         ExecuteQuery $readAction,
         ResponseFactoryInterface $responseFactory,
         string $routeName,
         UriGenerator $uriGenerator,
-        IdentifierGenerator $identifierGenerator
+        IdentifierGenerator $identifierGenerator,
+        RequestIdInjector $requestIdInjector
     ) {
         $this->writeAction         = $writeAction;
         $this->readAction          = $readAction;
         $this->responseFactory     = $responseFactory;
         $this->routeName           = $routeName;
         $this->uriGenerator        = $uriGenerator;
         $this->identifierGenerator = $identifierGenerator;
+        $this->requestIdInjector   = $requestIdInjector;
     }
 
     /**
      * {@inheritdoc}
      */
     public function handle(ServerRequestInterface $request): ResponseInterface
     {
-        $request = $request->withAttribute(
-            IdentifierGenerator::class,
-            $this->identifierGenerator->generate()
+        $request = $this->requestIdInjector->injectIdIfAbsent(
```

Comment:
I was hoping for something simpler here, like:

```php
$request = $request->withAttribute(
    IdentifierGenerator::class,
    $request->getAttribute(IdentifierGenerator::class, $this->identifierGenerator->generate())
);
```

Reply:
Nice, I wrote 200 lines of code to do something that could have been done in 3.